### PR TITLE
Add a script to check all models are tested and documented

### PR DESCRIPTION
**Pull request by sgugger**
This PR adds a new script to `make quality` that checks all models are tested and documented. This way, we will get a CI failure is someone adds a new model but doesn't document it or add it to the tests so that the common tests are applied to it. This will make the library far more robust.

Note: the changes in the doc files are related to models that were not documented. The changes in model files/test files are related to models that were not tested, had a bug and the corresponding fixes. I then got lazy and stopped trying to fix all the models not tested and just added TODO.

### CODE REVIEW

File: utils/check_repo.py [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+        if (
+            os.path.isfile(f"{PATH_TO_TESTS}/{filename}")
+            and filename.startswith("test_modeling")
+            and not filename[:-3] in _ignore_files
+        ):
+            test_files.append(filename)
+    return test_files
+
+
+# If some doc files should be ignored for the check models are documented, they should be added in the
```
Comment (sshleifer) [inline utils/check_repo.py:128]: I don't understand this comment

**Review by sshleifer — APPROVED** [review]
Love this!

File: Makefile [hunk 2]
```diff
@@ -6,6 +6,7 @@ quality:
 	black --check --line-length 119 --target-version py35 examples templates tests src utils
 	isort --check-only --recursive examples templates tests src utils
 	flake8 examples templates tests src utils
+	python utils/check_repo.py
```
Comment (sshleifer) [inline Makefile:9]: is it fast?

Comment (sgugger) [reply Makefile:9]: Takes less than a second, so faster than all the other three.

File: src/transformers/modeling_tf_albert.py [hunk 3]
```diff
@@ -1279,14 +1279,19 @@ def call(
...
+        )
 
         outputs = self.albert(
             flat_input_ids,
             flat_attention_mask,
             flat_token_type_ids,
             flat_position_ids,
             head_mask,
-            inputs_embeds,
+            flat_inputs_embeds,
```
Comment (JetRunner) [inline src/transformers/modeling_tf_albert.py:1294]: Isn't multiple choice working before this change?

File: utils/check_repo.py [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+# nested list _ignore_module of this function.
+def get_model_doc_files():
+    """ Get the model doc files."""
+    _ignore_files = [
```
Comment (JetRunner) [inline utils/check_repo.py:132]:
```suggestion
    _ignore_modules = [
```

File: utils/check_repo.py [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+        "auto",
+        "dialogpt",
+        "marian",
+        "retribert",
+    ]
+    doc_files = []
+    for filename in os.listdir(PATH_TO_DOC):
+        if os.path.isfile(f"{PATH_TO_DOC}/{filename}") and not filename[:-4] in _ignore_files:
```
Comment (JetRunner) [inline utils/check_repo.py:140]:
```suggestion
        if os.path.isfile(f"{PATH_TO_DOC}/{filename}") and not filename[:-4] in _ignore_modules:
```
(nitpick) Also, with `filename[:-4]` do you want to strip `.rst`? `os.path.splitext(filename)[0]` looks more readable to me.

File: src/transformers/modeling_tf_albert.py (hunk 3 above)
Comment (LysandreJik) [reply src/transformers/modeling_tf_albert.py:1294]: Probably not, not the TF version. That's usually what happens when models are added without tests 😕

**Review by LysandreJik — APPROVED** [review]
LGTM, thanks for doing this change!

File: utils/check_repo.py [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+
+# Update this list for models that are not tested with a comment explaining the reason it should not be.
+# Being in this list is an exception and should **not** be the rule.
+IGNORE_NON_TESTED = [
+    "BertLMHeadModel",  # Needs to be setup as decoder.
+    "DPREncoder",  # Building part of bigger (tested) model.
+    "DPRSpanPredictor",  # Building part of bigger (tested) model.
+    "ReformerForMaskedLM",  # Needs to be setup as decoder.
+    "T5Stack",  # Building part of bigger (tested) model.
+    "TFAlbertForMultipleChoice",  # TODO: fix
```
Comment (LysandreJik) [inline utils/check_repo.py:21]: fixing

Comment (LysandreJik) [inline utils/check_repo.py:21]: This one is fixed already, right?

File: utils/check_repo.py [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+    "TFAlbertForTokenClassification",  # TODO: fix
```
Comment (LysandreJik) [inline utils/check_repo.py:22]: Fixing

File: utils/check_repo.py [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+    "TFBertLMHeadModel",  # TODO: fix
```
Comment (LysandreJik) [inline utils/check_repo.py:23]: Fixing

File: utils/check_repo.py [hunk 9]
```diff
@@ -0,0 +1,269 @@
...
+    "TFElectraForMultipleChoice",  # Fix is in #6284
+    "TFElectraForQuestionAnswering",  # TODO: fix
```
Comment (LysandreJik) [inline utils/check_repo.py:25]: Fixing

File: utils/check_repo.py (hunk 6 above)
Comment (sgugger) [reply utils/check_repo.py:21]: No I fixed half of it and then gave up for the other half ;-)

**Pull request merged**